### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by airportyh**
Added tests for count. And made API docs' parameters more consistent.

### CODE REVIEW

File: test/enumerable.js [hunk 1]
```diff
@@ -302,6 +302,18 @@ describe('enumerable', function () {
     });
   });
 
+  describe('count', function () {
+    it('counts w function', function () {
```
Comment (Swatinem) [inline test/enumerable.js]: you can write out `with` :-)

File: test/enumerable.js [hunk 2]
```diff
@@ -302,6 +302,18 @@ describe('enumerable', function () {
...
+      assert(1 === fruits.count(function(f){
+        return f.name === 'grape'
+      }));
+    });
+
+    it('counts w string', function () {
```
Comment (Swatinem) [inline test/enumerable.js]: same

**Comment by airportyh** [comment]
Cool! Would you be interested in PRs for the following as well?
- indexOf - doesn't handle second parameter fromIndex.
- iteration methods (map, forEach, reduce, etc) don't pass 3rd parameter (array) to callback.
- iteration methods don't set context for the callback when provided as last parameter.

These are for EcmaScript 5 compatibility. I could also run a full EcmaScript 5 compatibility test suite against it to see what else I find.

**Comment by Swatinem** [comment]
Of course I would :-)

#map() and some others are built to recreate the array instance underneath.
Apart from those, all we really do is provide wrappers that work with `to-function` so you can pass in a simple string.
Might as well make a simple wrapper generator that does the `to-function` magic and then delegates to the native Array.prototype methods.

**Pull request merged**